packages/muelu/src/Rebalancing/MueLu_RepartitionFactory_def.hpp

```diff
@@ -101,12 +101,14 @@ namespace MueLu {
     Input(currentLevel, "Partition");
   }
 
-  template<class T> class MpiTypeTraits            { public: static MPI_Datatype getType(); };
-  template<>        class MpiTypeTraits<long>      { public: static MPI_Datatype getType() { return MPI_LONG;      } };
-  template<>        class MpiTypeTraits<int>       { public: static MPI_Datatype getType() { return MPI_INT;       } };
-  template<>        class MpiTypeTraits<short>     { public: static MPI_Datatype getType() { return MPI_SHORT;     } };
-  template<>        class MpiTypeTraits<unsigned>  { public: static MPI_Datatype getType() { return MPI_UNSIGNED;  } };
-  template<>        class MpiTypeTraits<long long> { public: static MPI_Datatype getType() { return MPI_LONG_LONG; } };
+  template<class T> class MpiTypeTraits                     { public: static MPI_Datatype getType(); };
```

Comment:
Doesn't Teuchos also define `MpiTypeTraits`?  The problem might already be solved there.

Reply:
It may. Removing `MpiTypeTraits` may be feasible, but out of scope for what I'm trying to do right now.

Reply:
Actually, nevermind. I may just go ahead and use Teuchos. I guess it should be fine to use `Teuchos::Details` namespace as `MpiTypeTraits` are not really exposed in Teuchos proper.

Reply:
@aprokop It's OK, I don't mean to hinder the PR.  If you're done pushing commits, I'll approve this one.  Thanks!

Reply:
Ah, I see, Christian approved it already :-)